File: automation-test-selenium/src/main/java/org/suntower/pages/core/AdminShellPage.java
package org.suntower.pages.core;

import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class AdminShellPage extends BasePage {
  protected String path;

  public AdminShellPage(WebDriver driver) {
    super(driver);
  }

  public void open() {
    if (path == null || path.isBlank()) {
      throw new IllegalStateException("This admin page does not define a path.");
    }
    visit(path);
  }

  public void goToBuildings() {
    click(linkByHref("/admin/building/list"));
  }

  public void goToCustomers() {
    click(linkByHref("/admin/customer/list"));
  }

  public void goToContracts() {
    click(linkByHref("/admin/contract/list"));
  }

  public void goToSaleContracts() {
    click(linkByHref("/admin/sale-contract/list"));
  }

  public void goToInvoices() {
    click(linkByHref("/admin/invoice/list"));
  }

  public void goToReports() {
    click(linkByHref("/admin/report"));
  }

  public void goToStaffs() {
    click(linkByHref("/admin/staff/list"));
  }

  public void goToProfile() {
    click(linkByHref("/admin/profile"));
  }
}

File: automation-test-playwright/pages/core/AdminShellPage.ts
import { BasePage } from "./BasePage";

export class AdminShellPage extends BasePage {
  protected readonly path?: string;

  /**
   * Opens the page.
   */
  async open(): Promise<void> {
    if (!this.path) {
      throw new Error("This admin page does not define a path.");
    }

    await this.visit(this.path);
  }

  /**
   * Navigates to buildings.
   */
  async goToBuildings(): Promise<void> {
    await this.linkByHref("/admin/building/list").click();
  }

  /**
   * Navigates to customers.
   */
  async goToCustomers(): Promise<void> {
    await this.linkByHref("/admin/customer/list").click();
  }

  /**
   * Navigates to contracts.
   */
  async goToContracts(): Promise<void> {
    await this.linkByHref("/admin/contract/list").click();
  }

  /**
   * Navigates to sale contracts.
   */
  async goToSaleContracts(): Promise<void> {
    await this.linkByHref("/admin/sale-contract/list").click();
  }

  /**
   * Navigates to invoices.
   */
  async goToInvoices(): Promise<void> {
    await this.linkByHref("/admin/invoice/list").click();
  }

  /**
   * Navigates to reports.
   */
  async goToReports(): Promise<void> {
    await this.linkByHref("/admin/report").click();
  }

  /**
   * Navigates to staffs.
   */
  async goToStaffs(): Promise<void> {
    await this.linkByHref("/admin/staff/list").click();
  }

  /**
   * Navigates to profile.
   */
  async goToProfile(): Promise<void> {
    await this.linkByHref("/admin/profile").click();
  }
}
